client/tailscale/tailscale.go

```diff
@@ -72,13 +71,12 @@ func (c *Client) httpClient() *http.Client {
 // For example, BuildURL(devices, 5) with the default server URL would result in
 // https://api.tailscale.com/api/v2/devices/5.
 func (c *Client) BuildURL(pathElements ...any) string {
-	elem := make([]string, 2, len(pathElements)+1)
-	elem[0] = c.baseURL()
-	elem[1] = "/api/v2"
+	elem := make([]string, 1, len(pathElements)+1)
+	elem[0] = "/api/v2"
 	for _, pathElement := range pathElements {
-		elem = append(elem, url.PathEscape(fmt.Sprint(pathElement)))
+		elem = append(elem, fmt.Sprint(pathElement))
```

Comment:
Don't we still want the escaping to apply here? (We add the "/" separators later, so they will still be safe)

Reply:
I was torn on this. This method is fairly new: https://github.com/tailscale/tailscale/commit/1f1a26776bc182652cfa9a4dc2a2c60b1b5de8b8, and before it we didn't escape anything in the path.
But it feels like we should, for some path elements. I can maybe over-engineer it with some special `client.BuildTailnetURL(client.DoNotEscape("acl?details=1"))` wrapper type.

Reply:
I agree we should escape if we can. I don't think we need to add an escape-escape hatch. 


My intuition is that a circumstance where someone wants to be clever, they can still build the base URL (by calling this with no arguments) and then do their own path surgery. (That might, perhaps, also discourage people from doing that, which would be somewhat weird given it's our APIs)

Reply:
Added an escape hatch, but I'm also open to having the caller modify URL manually. WDYT?

Reply:
> but I'm also open to having the caller modify URL manually. WDYT

I'd prefer that over the new type.